sdk/test-utils/recorder/src/utils/index.ts

```diff
@@ -101,6 +101,24 @@ export function encodeRFC3986(str: string): string {
   );
 }
 
+/**
+ * RFC 3986 specification doesn't take `~` character into account.
```

Comment:
For the record, RFC 3986 says:

> Unreserved Characters
> 
> Characters that are allowed in a URI but do not have a reserved
> purpose are called unreserved.  These include uppercase and lowercase
> letters, decimal digits, hyphen, period, underscore, and tilde.
> 
> unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"

Later it says:

> When a URI is dereferenced, the components and subcomponents
> significant to the scheme-specific dereferencing process (if any)
> must be parsed and separated before the percent-encoded octets within
> those components can be safely decoded, as otherwise the data may be
> mistaken for component delimiters.  The only exception is for
> percent-encoded octets corresponding to characters in the unreserved
> set, which can be decoded at any time.  For example, the octet
> corresponding to the tilde ("\~") character is often encoded as "%7E"
> by older URI processing implementations; the "%7E" can be replaced by
> "\~" without changing its interpretation.

This specification was added in 2005. JS encoding methods are likely older (but there could be some other explanation of why this is missing that I might not be aware of).

Reply:
since the specification says that these are the characters they ignore: "-" / "." / "_" / "~",

let’s make sure we have a test in which we check how bad things get if we have a secret with those other characters 🙌